coderd/database/queries/workspaces.sql

```diff
@@ -161,17 +162,63 @@ WHERE
 			name ILIKE '%' || @name || '%'
 		ELSE true
 	END
+	-- Filter by agent status
+	-- has-agent: is only applicable for workspaces in "start" transition. Stopped and deleted workspaces don't have agents.
+	AND CASE
+		WHEN @has_agent :: text != '' THEN
+			(
+				SELECT COUNT(*)
+				FROM
+					workspace_builds
+				JOIN
+					provisioner_jobs
+				ON
+					provisioner_jobs.id = workspace_builds.job_id
+				JOIN
+					workspace_resources
+				ON
+					workspace_resources.job_id = provisioner_jobs.id
```

Comment:
I haven't tested if this works, but I would suggest this final change:

```suggestion
				FROM
					latest_build
				JOIN
					workspace_resources
				ON
					workspace_resources.job_id = provisioner_job_id
```

You'll need to add `provisioner_jobs.id AS provisioner_job_id ` to the select for `latest_build` though.

This should be more performant than re-joining these tables here. (It also eliminates the need for the `build_number` check in `WHERE`.)

Reply:
I'm afraid that it would work if `latest_build` is CTE. Otherwise, it's: `relation "latest_build" does not exist`.